server/admin/prompt.go

```diff
@@ -12,26 +13,53 @@ import (
 	"golang.org/x/crypto/ssh/terminal"
 )
 
+type passwordReadResult struct {
+	password []byte
+	err      error
+}
+
 // PasswordPrompt prompts the user to enter a password. Password must not be an
 // empty string.
-func PasswordPrompt(prompt string) ([]byte, error) {
-	fmt.Print(prompt)
-	pass, err := terminal.ReadPassword(syscall.Stdin)
-	fmt.Println()
+func PasswordPrompt(ctx context.Context, prompt string) ([]byte, error) {
+	// Get the initial state of the terminal.
+	initialTermState, err := terminal.GetState(syscall.Stdin)
 	if err != nil {
 		return nil, err
 	}
-	if pass == nil {
-		return nil, errors.New("password must not be empty")
+
+	done := make(chan passwordReadResult, 1)
+
+	fmt.Print(prompt)
+	go func() {
+		pass, err := terminal.ReadPassword(syscall.Stdin)
+		done <- passwordReadResult{
+			password: pass,
+			err:      err,
+		}
+	}()
+
+	select {
+	case <-ctx.Done():
+		_ = terminal.Restore(syscall.Stdin, initialTermState)
```

Comment:
Can you log this returned error?

Reply:
There's no logger in the `server/admin` package. Do we need to concern ourselves with any error returned from this call though? I think main op here is returning when ctx is cancelled without waiting for `terminal.ReadPassword` to return.

Reply:
I'm fine with no error here since a non-nil error will make it back to the caller as @itswisdomagain said.  Although if there is an error I have no idea from `Restore`, it should not be common and it can't hurt to simply `fmt.Println(err)` if it's non-nil so we can learn something from it even if it tells the user little.

Also I noticed the docs for the `terminal` package have an example:

```go
oldState, err := terminal.MakeRaw(0)
if err != nil {
        panic(err)
}
defer terminal.Restore(0, oldState)
```

These examples are obv terse, but I get the feeling there's not much to be done with the error from Restore.